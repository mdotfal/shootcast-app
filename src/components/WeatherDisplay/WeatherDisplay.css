.weather-display {
  display: flex;
  flex-direction: row;
}

.weather-display h2 {
  margin: 5px;
}

.weather-display-items {
  display: flex;
  justify-content: flex-start;
  flex-direction: row;
  align-items: center;
  padding: 0 10px;
  width: 250px;
}

.current-weather-forecast span {
  padding: 5px;
}
