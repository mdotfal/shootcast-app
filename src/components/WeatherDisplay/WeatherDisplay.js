import React from 'react';
import './WeatherDisplay.css';

function WeatherDisplay() {
  return (
    <section>
      <div className="current-weather-display">
        <p>Sunny</p>
      </div>

      <div className="current-weather-forecast">
        <span>Sunday</span>
        <span>Monday</span>
        <span>Tuesday</span>
        <span>Wednesday</span>
        <span>Thursday</span>
        <span>Friday</span>
        <span>Saturday</span>
      </div>
    </section>
  );
}

export default WeatherDisplay;