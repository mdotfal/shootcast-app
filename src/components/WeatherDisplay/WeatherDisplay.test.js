import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WeatherDisplay from './WeatherDisplay';

describe( `WeatherDisplay component`, () => {
  const data = {
    dt: 'test dt',
    main: {
      temp: 50
    },
    weather: [
      { description: "Test Description" }
    ]
  }

  it( `renders without crashing`, () => {
    const div = document.createElement( 'div' );
    ReactDOM.render( <WeatherDisplay />, div );
    ReactDOM.unmountComponentAtNode( div ); 
  })
  
  it( `renders with data`, () => {
    const wrapper = shallow( <WeatherDisplay data={ data }/> )
    expect( toJson( wrapper )).toMatchSnapshot()
  })

  it( `renders UI as expected`, () => {
    const wrapper = shallow( <WeatherDisplay /> )
    expect( toJson( wrapper )).toMatchSnapshot()
  })

})