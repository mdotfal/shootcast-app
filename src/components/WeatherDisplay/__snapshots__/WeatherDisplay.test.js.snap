// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDisplay component renders UI as expected 1`] = `
<div
  className="weather-display"
>
  <h2>
    NaN
    °F
  </h2>
  <div
    className="weather-display-items"
  >
    <div
      className="current-weather-display"
    >
      Invalid date
    </div>
     | 
    <div
      className="current-weather-description"
    />
  </div>
</div>
`;

exports[`WeatherDisplay component renders with data 1`] = `
<div
  className="weather-display"
>
  <h2>
    50
    °F
  </h2>
  <div
    className="weather-display-items"
  >
    <div
      className="current-weather-display"
    >
      Invalid date
    </div>
     | 
    <div
      className="current-weather-description"
    >
      Test Description
    </div>
  </div>
</div>
`;
